code/modules/research/xenobiology/crossbreeding/warping_runes.dm

```diff
@@ -0,0 +1,309 @@
+///////////////////////////////////////////////
+//////////////////SLIME RUNES//////////////////
+///////////////////////////////////////////////
+
+/obj/effect/slimerune/grey
+	name = "grey rune"
+	desc = "The center has an outline of a slime extract in it."
+
+/obj/effect/slimerune/grey/examine(mob/user)
+	..()
+	var/obj/item/slimecross/warping/grey/ex = extract
+	to_chat(user, "<span class='notice'>It has absorbed [ex.num_absorbed] extract[(ex.num_absorbed != 1) ? "s":""].</span>")
+	to_chat(user, "<span class='notice'>At 8 extracts, it can be manually fed an extract to produce a slime of that color.</span>")
+
+/obj/effect/slimerune/grey/process()
+	var/obj/item/slimecross/warping/grey/ex = extract
+	if(ex.num_absorbed >= 8)
+		return
+	if(locate(/obj/item/slime_extract) in get_turf(loc))
+		var/eaten = 0
+		for(var/obj/item/slime_extract/x in get_turf(src))
+			if(ex.num_absorbed >= 8)
+				break
+			qdel(x)
+			ex.num_absorbed++
+			eaten += 1
+		visible_message("<span class='notice'>[src] glows, and the extract[eaten != 1 ? "s":""] on it fade[eaten != 1 ? "":"s"] away.</span>")
+		if(ex.num_absorbed >= 8)
+			visible_message("<span class='notice'>[src] has finished its bluespace calibration, and can now produce a slime.</span>")
+
+/obj/effect/slimerune/grey/attackby(obj/item/slime_extract/O, mob/user)
```

Comment:
Prefer to use the actual type that can possibly come in inside proc arguments. Recast it